columnflow/tasks/framework/mixins.py

```diff
@@ -691,10 +691,35 @@ def store_parts(self) -> law.util.InsertableDict:
         # since MLTraining is no CalibratorsMixin, SelectorMixin, ProducerMixin, ConfigTask,
         # all these parts are missing in the `store_parts`
 
-        configs_repr = "__".join(self.configs)
+        configs_repr = "__".join(self.configs[:5])
+
+        if len(self.configs) > 5:
+            configs_repr += f"_{law.util.create_hash(self.configs[5:])}"
+
         parts.insert_after("task_family", "configs", configs_repr)
 
-        # TODO: parts for calib, sel, prod
+        task_array_functions = {
+            "calib": self.calibrators,
+            "sel": tuple((sel,) for sel in self.selectors),
+            "prod": self.producers,
+        }
+
+        for label, fct_names in task_array_functions.items():
+            if all(map(lambda x: x == fct_names[0], fct_names)):
```

Comment:
Not that important, but doing `fct_names[0]` could raise an `IndexError` if empty lists are passed for `self.configs`. Is this caught somewhere?

Reply:
Yes, you can't run a MLModel without specifying some config (otherwise you get this Exception)
```
Exception: MLModel 'base_test' received configs '()' to define training configs, but did not define any
```

Reply:
It seems that any `[][0]` would be caught here as well now, so resolving :)